surface/include/pcl/surface/impl/mls.hpp

```diff
@@ -116,24 +116,30 @@ pcl::MovingLeastSquares<PointInT, PointOutT>::process (PointCloudOut &output)
       float tmp = static_cast<float> (search_radius_ / 2.0f);
       boost::uniform_real<float> uniform_distrib (-tmp, tmp);
       rng_uniform_distribution_.reset (new boost::variate_generator<boost::mt19937&, boost::uniform_real<float> > (rng_alg_, uniform_distrib));
-
-      mls_results_.resize (1); // Need to have a reference to a single dummy result.
       
       break;
     }
     case (VOXEL_GRID_DILATION):
     case (DISTINCT_CLOUD):
       {
-        mls_results_.resize (input_->size ());
+        cache_mls_results_ = true;
```

Comment:
So if the method is VOXEL_GRID_DILATION or DISTINCT_CLOUD, the caching is automatically set to true, despite if the user previously set caching to be false. It this happens we should the throwing a warning.

For everything else, it uses whatever option the user has decided before.